lib/compress/zstd_compress.c

```diff
@@ -2321,8 +2321,10 @@ size_t ZSTD_getSequences(ZSTD_CCtx* zc, ZSTD_Sequence* outSeqs,
 {
     const size_t dstCapacity = ZSTD_compressBound(srcSize);
     void* dst = ZSTD_malloc(dstCapacity, ZSTD_defaultCMem);
-
     SeqCollector seqCollector;
+
+    assert(dst != NULL);
```

Comment:
An `assert()` means that this situation should _never_ happen.
It's only activated during debug sessions, to ensure that the condition is indeed valid as promised, and then is removed at production time.

In this case, we can't _guarantee_ that `malloc()` will necessarily succeed.
It may fail and return `NULL`.

Therefore, what's needed here is to actively check this condition, and _return an __error code___ when it's triggered.

`memory_allocation` seems the correct error code.
`RETURN_ERROR_IF()` seems the right macro to deal with this situation.